Frameworks/Foundation/NSDictionary.mm

```diff
@@ -99,8 +99,11 @@ - (NSArray*)allObjects {
         return nil;
     }
 
-    StrongId<NSArray> keyArray = [_keyEnumerator allObjects];
-    StrongId<NSMutableArray> toRet = [NSMutableArray new];
+    StrongId<NSArray> keyArray;
+    keyArray.attach([_keyEnumerator allObjects]);
+
+    StrongId<NSMutableArray> toRet;
```

Comment:
yes. Its a good habit.  Also [Waffle new] is *not* autoreleased.

---
In reply to: [73973293](https://github.com/Microsoft/WinObjC/pull/851#discussion_r73973293) [](ancestors = 73973293)